trin-core/src/jsonrpc/types.rs

```diff
@@ -460,17 +464,21 @@ impl<TContentKey: OverlayContentKey> TryFrom<[&Value; 2]> for StoreParams<TConte
         let content_key = params[0]
             .as_str()
             .ok_or_else(|| ValidationError::new("Empty content key param"))?;
-        let content_key = match hex::decode(content_key) {
+        let content_key = match hex_decode(content_key) {
             Ok(val) => match TContentKey::try_from(val) {
                 Ok(val) => val,
                 Err(_) => return Err(ValidationError::new("Unable to decode content_key")),
             },
-            Err(_) => return Err(ValidationError::new("Unable to decode content_key")),
+            Err(_) => {
```

Comment:
Hm, I had a fight with `ValidationError` library, and it nearly won. It was telling me:
```rust
474 |                     format!("Cannot convert content_key hex to bytes: {}", err),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found struct `std::string::String`
```
and providing a reference didn't work, because the `String` would drop out of scope at return, while the `ValidationError` was still holding it. So I come up with this ugly monster after digging into the VE docs:
```rs
                let mut ve = ValidationError::new("Cannot convert content_key hex to bytes");
                ve.add_param(std::borrow::Cow::Borrowed("hex_decode_exception"), &err.to_string());
                return Err(ve);
```